### PR TITLE
Ipfs refacto

### PR DESCRIPTION
**Pull request by salimtb**
The purpose of this PR is to remove the use of the ipfs-http-client module and replace it with http calls to the ipfs public api.
the package is not compatible with webpack 5, so that's why we replace it

### CODE REVIEW

File: e2e/utils/utils.ts/ipfsServerClient.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+          'Content-Type': `multipart/form-data; boundary=${formData.getBoundary()}`,
+        },
+      });
+
+      const arr = response.data.split('\n').filter(Boolean).map(JSON.parse);
+      const lastItem = arr.pop();
+      return lastItem.Hash;
+    } catch (error) {
+      return log.throwArgumentError(Logger.message.an_error_occured_with_ipfs_api, 'file', files, {
+        location: Logger.location.IPFSSERVICE_UPLOADFILE,
```
Comment (sahar-fehri) [inline e2e/utils/utils.ts/ipfsServerClient.ts:86]: I think here for the location you meant to put:` Logger.location.IPFS_SERVER_UPLOADDIRECTORY `, ?

File: src/lib/Logger/index.ts [hunk 2]
```diff
@@ -37,6 +37,13 @@ export enum ErrorLocation {
   IPFSSERVICE_UPLOADDIRECTORY = '[IPFS.uploadDirectory]',
   IPFSSERVICE_UNPINFILE = '[IPFS.unPinFile]',
 
+  // IPFS SERVER CLIENT
+  IPFS_SERVER_CONSTRUCTOR = '[IPFS.constructor]',
+  IPFS_SERVER_UPLOADFILE = '[IPFS.uploadFile]',
+  IPFS_SERVER_UPLOADOBJECT = '[IPFS.uploadObject]',
```
Comment (sahar-fehri) [inline src/lib/Logger/index.ts:43]: where is the IPFS_SERVER_UPLOADOBJECT  used?

Comment (salimtb) [reply src/lib/Logger/index.ts:43]: no where , we don't need it , i'll remove it now

File: integration-test/sdk.nfts.search.test-integration.ts [hunk 3]
```diff
@@ -8,6 +8,8 @@ loadEnv();
 
 describe('E2E Test: Sdk (read)', () => {
   let sdk: SDK;
+  jest.setTimeout(120 * 1000);
```
Comment (sahar-fehri) [inline integration-test/sdk.nfts.search.test-integration.ts:11]: why did we have to add this timeout?

File: src/lib/Logger/index.ts (hunk 2 above)
Comment (sahar-fehri) [reply src/lib/Logger/index.ts]: 🙏

**Review by sahar-fehri — APPROVED** [review]

File: src/services/ipfsService.ts [hunk 4]
```diff
@@ -92,41 +91,18 @@ export default class IPFS {
...
-      return `ipfs://${item?.cid.toString()}/`;
-    } catch (error) {
-      return log.throwError(Logger.message.an_error_occured_with_ipfs_api, Logger.code.IPFS, {
-        location: Logger.location.IPFSSERVICE_UPLOADDIRECTORY,
-        error,
       });
-    }
-  }
-
-  async unPinFile({ hash }: { hash: string }) {
```
Comment (bergeron) [inline src/services/ipfsService.ts:118]: It looks like this `unPinFile` function was removed.  Does it need to stay?

Comment (salimtb) [reply src/services/ipfsService.ts:118]: we don't expose it through the main file ( index.js ) , so even with the old version this function wasn't available for users

**Review by bergeron — APPROVED** [review]

**Pull request merged**